continuous_integration/kubernetes/helm-lint-and-template.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -e
+
+this_dir="$(dirname "${BASH_SOURCE[0]}")"
+git_root="$(cd "${this_dir}/../.." && pwd)"
+
+echo "Linting Helm Chart"
+helm lint \
```

Comment:
I wonder if we could use precommit here instead of a CI script to keep things consistent with `dask/helm-chart`.

https://github.com/dask/helm-chart/blob/main/.pre-commit-config.yaml#L6-L9

Reply:
From inspection, the referenced GitHub action to do `helm lint` [don't support referencing our test config in the current location](https://github.com/gruntwork-io/pre-commit/blob/e9250bd69bb312d55364213ff5ff037a09be55d9/hooks/helmlint.sh#L113-L120).

I suggest if there is a pre-commit hook to be added about helm lint, that it is done in a separate PR in addition to this which references a specific config file referenced by other tests.

Reply:
> I suggest if there is a pre-commit hook to be added about helm lint, that it is done in a separate PR in addition to this which references a specific config file referenced by other tests.

I understand where you are coming from, but this PR adds a listing script, and it would be preferable to do this via precommit to be consistent with other places.

If we bump precommit to a separate PR then that PR would also remove this script right?

Reply:
This script includes `helm lint` and `helm template`. The key thing I wanted to add was actually `helm template` which is what I find to typically catch most template issues. So by using the dedicated `helm lint` pre-commit action by gruntworks-io and making pre-commit run as part of the CI system, it would not solve the wish to have a `helm template` step here.

Here are options I could execute on directly:
1. Remove `helm lint` from this script:
   a) and add a pre-commit hook to do `helm lint`
   b) and don't do `helm lint` in the CI system
2. Remove this script entirely and do a local pre-commit hook to do both `helm template` as well as `helm lint`
3. Stick to not referencing `helm` in pre-commit but instead doing it only in the GitHub Workflows

I generally dislike the complexity added by using a third party action to do something as minor as running `helm lint` because of how it makes it harder to understand whats going on, so in my order of preference if we are to use pre-commit would be: 2 -> 1b -> 1a, but due to an experience described below, I suggest 3.

### Using pre-commit.ci

In the JupyterHub org, we have went all out on pre-commit and started using also the pre-commit.ci service. It helps to:
- update versions referenced in pre-commit-config.yaml files
- push pre-commit changes to PRs that fail tests
- run pre-commit checks in a very quick and performant manner and report its status like any other GitHub workflow job

If pre-commit need to reference the `helm` binary directly, as `helm template` would, it would mean that there is a dependency on that binary for CI systems to run. Local users would need to install `helm` even though they don't work with the Helm chart but instead work on dask-gateway's Python code base, and pre-commit.ci wouldn't be able to handle this either.

For reasons like this, we have [explicitly removed `helm` in the JupyterHub helm chart's pre-commit config](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/commit/c37d5f79cb838df5c8e94677f8f6b5b0a27fb804#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9). Even if we would reference gruntworks-io, it [would require `helm` to be available](https://github.com/gruntwork-io/pre-commit/blob/e9250bd69bb312d55364213ff5ff037a09be55d9/hooks/helmlint.sh#L122-L132) and fail without it being installed.

---

@jacobtomlinson let me know what you would like me to do and I'll get it done!

Reply:
Thanks for taking the time to write this up @consideRatio. That's compelling enough for me. Let's go with this as is.

Reply:
Thanks for the review and patience with me on this @jacobtomlinson!